ocaml/tests/test_observer.ml

```diff
@@ -262,32 +265,28 @@ let clear_export_dir () =
   with _ -> ()
 
 let verify_json_fields_and_values ~json =
-  let open Yojson.Basic.Util in
-  try
-    let tags = json |> member "tags" in
-    let _ = tags |> member "xs.pool.uuid" |> to_string in
-    let _ = tags |> member "xs.host.name" |> to_string in
-    let _ = tags |> member "xs.host.uuid" |> to_string in
-    let observer_name = tags |> member "xs.observer.name" |> to_string in
-    Alcotest.(check string)
-      "Observer name in trace json as expected" "test-observer" observer_name ;
-
-    let local_endpoint = json |> member "localEndpoint" in
-    let service_name = local_endpoint |> member "serviceName" |> to_string in
-    Alcotest.(check string)
-      "Service name in trace json as expected" "xapi" service_name ;
-
-    let _ = json |> member "duration" |> to_int in
-    let _ = json |> member "timestamp" |> to_int in
-
-    let name = json |> member "name" |> to_string in
-    Alcotest.(check string)
-      "Task name in trace json as expected" "test_task" name ;
-
-    let _ = json |> member "traceId" |> to_string in
-    let _ = json |> member "id" |> to_string in
-    ()
-  with e -> Alcotest.failf "%s" (Printexc.to_string e)
+  match json with
+  | `Assoc
+      [
+        ( "tags"
+        , `Assoc
+            [
+              ("xs.pool.uuid", `String _)
+            ; ("xs.host.name", `String _)
+            ; ("xs.host.uuid", `String _)
+            ; ("xs.observer.name", `String "test-observer")
+            ]
+        )
+      ; ("localEndpoint", `Assoc [("serviceName", `String "xapi")])
+      ; ("duration", `Int _)
+      ; ("timestamp", `Int _)
+      ; ("name", `String "test_task")
+      ; ("traceId", `String _)
+      ; ("id", `String _)
+      ] ->
+      ()
+  | _ ->
+      Alcotest.failf "Trace log json format incorrect"
```

Comment:
If this ever fails you will want to see the actual JSON. Would turn the JSON into a string and include it here.